### PR TITLE
Es improvements

### PR DESCRIPTION
**Pull request by LourensVeen**
And a comment, see inline.

### CODE REVIEW

File: xtas/tasks/es.py [hunk 1]
```diff
@@ -100,32 +139,58 @@ def fetch_query_batch(idx, typ, query, field='body'):
     hits = scan(_es(), {'query': query}, index=idx, doc_type=typ)
     return [hit['_source'][field] for hit in hits if field in hit['_source']]
 
-CHECKED_MAPPINGS = set()
```
Comment (LourensVeen) [inline xtas/tasks/es.py]: Does this cache disappear between sessions, while the types remain in the ES instance? Does it really help that much, considering that it's only used after a (presumably somewhat lengthy) calculation? Should it be named _CHECKED_MAPPINGS?

Comment (larsmans) [inline xtas/tasks/es.py]: Yes, the types remain while the cache gets flushed. I didn't write or profile this, but I imagine that doing a put_mapping for each document can get quite expensive.

**Comment by larsmans** [comment]
Could you rebase d63a3ee43a4492db4c7a8df0f6a91099c348c679 out of this branch?

**Comment by LourensVeen** [comment]
Like this, or do you want a rebase to current master too? Doesn't look like there are conflicts.

**Comment by larsmans** [comment]
Looks fine.

File: xtas/tasks/es.py (hunk 1 above)
Comment (LourensVeen) [inline xtas/tasks/es.py]: Well, but it queries ES anyway, and only adds the mapping if it's not already there. So the only thing that's being saved is a single ES mapping query per result. And I think it should probably be private, but that's issue #111 already, so I'll leave that out of this one.

**Pull request merged**